### PR TITLE
Hoodie rework & resprite

### PR DESCRIPTION
**Pull request by VTCobaltblood**
Hoodies have been resprited to be less ugly, and now also function in a similar way to winter coats, i.e. actually having a hood. In addition to that, they can still be opened and closed. They're also recolorable in the loadout.
The hood icon not being colored is a known issue, but I'm not sure how to fix it. This affects roughly nothing except for the UI.
The PR also makes a slight tweak to my cardigan sprite for it to be less ugly.

### CODE REVIEW

**Comment by VTCobaltblood** [comment]
!review

File: code/modules/clothing/suits/hoodies.dm [hunk 1]
```diff
@@ -174,3 +174,69 @@
...
+	icon_state = "hoodie"
+	var/icon_open = "hoodie_open"
+	var/icon_closed = "hoodie"
+	item_state = "hoodie"
+	hoodtype = /obj/item/clothing/head/winterhood/hoodie
+	/obj/item/clothing/suit/storage/hooded/wintercoat/hoodie/RemoveHood()
+		..()
+		icon_open = "[initial(icon_open)]"
+		icon_closed = "[initial(icon_closed)]"
+	verb/Toggle() //copied from storage toggle
```
Comment (Alberyk) [inline code/modules/clothing/suits/hoodies.dm]: Don't use relative pathing.

File: code/modules/clothing/suits/hoodies.dm [hunk 2]
```diff
@@ -174,3 +174,69 @@
...
+			to_chat(usr, "You zip the hoodie.")
+		else if(icon_state == icon_closed)
+			icon_state = icon_open
+			item_state = icon_open
+			to_chat(usr, "You unzip the hoodie.")
+		else
+			to_chat(usr, "You attempt to zip the velcro on your [src], before promptly realising how silly you are.")
+			return
+		update_clothing_icon()
+	verb/ToggleHoodie()
```
Comment (Alberyk) [inline code/modules/clothing/suits/hoodies.dm]: Don't use relative pathing.

File: code/modules/clothing/suits/hoodies.dm [hunk 3]
```diff
@@ -174,3 +174,69 @@
...
+	hoodtype = /obj/item/clothing/head/winterhood/hoodie
+	/obj/item/clothing/suit/storage/hooded/wintercoat/hoodie/RemoveHood()
+		..()
+		icon_open = "[initial(icon_open)]"
+		icon_closed = "[initial(icon_closed)]"
+	verb/Toggle() //copied from storage toggle
+		set name = "Toggle Coat Buttons"
+		set category = "Object"
+		set src in usr
+		if(!usr.canmove || usr.stat || usr.restrained())
```
Comment (skull132) [inline code/modules/clothing/suits/hoodies.dm]: Use `can_use(usr)` please and shank you.

File: code/modules/clothing/suits/hoodies.dm [hunk 4]
```diff
@@ -174,3 +174,69 @@
...
+		else
+			to_chat(usr, "You attempt to zip the velcro on your [src], before promptly realising how silly you are.")
+			return
+		update_clothing_icon()
+	verb/ToggleHoodie()
+		set name ="Toggle Coat Hood"
+		set category = "Object"
+		set src in usr
+
+		if(!usr.canmove || usr.stat || usr.restrained())
```
Comment (skull132) [inline code/modules/clothing/suits/hoodies.dm]: Use `can_use(usr)` please and shank you.

File: code/modules/clothing/suits/hoodies.dm [hunk 5]
```diff
@@ -174,3 +174,69 @@
...
+			return 0
+
+		if(!suittoggled)
+			if(ishuman(loc))
+				var/mob/living/carbon/human/H = src.loc
+				if(H.wear_suit != src)
+					H << "<span class='warning'>You must be wearing [src] to put up the hood!</span>"
```
Comment (skull132) [inline code/modules/clothing/suits/hoodies.dm]:
```suggestion
					to_chat(H, "<span class='warning'>You must be wearing [src] to put up the hood!</span>")
```

File: code/modules/clothing/suits/hoodies.dm [hunk 6]
```diff
@@ -174,3 +174,69 @@
...
+					return
+				if(H.head)
+					H << "<span class='warning'>You're already wearing something on your head!</span>"
```
Comment (skull132) [inline code/modules/clothing/suits/hoodies.dm]:
```suggestion
					to_chat(H, "<span class='warning'>You're already wearing something on your head!</span>")
```

File: code/modules/clothing/suits/hoodies.dm [hunk 7]
```diff
@@ -174,3 +174,69 @@
...
+
+/obj/item/clothing/suit/storage/hooded/wintercoat/hoodie
+	name = "hoodie"
+	desc = "Warm and cozy."
+	icon_state = "hoodie"
+	var/icon_open = "hoodie_open"
+	var/icon_closed = "hoodie"
+	item_state = "hoodie"
+	hoodtype = /obj/item/clothing/head/winterhood/hoodie
+	/obj/item/clothing/suit/storage/hooded/wintercoat/hoodie/RemoveHood()
```
Comment (Karolis2011) [inline code/modules/clothing/suits/hoodies.dm]: Why this and following lines are indented so strangely.

File: code/modules/clothing/suits/hoodies.dm [hunk 8]
```diff
@@ -174,3 +174,69 @@
...
+		..()
+		icon_open = "[initial(icon_open)]"
+		icon_closed = "[initial(icon_closed)]"
```
Comment (Karolis2011) [inline code/modules/clothing/suits/hoodies.dm]: Is there a need to encapsulate initial inside string template?

Comment (VTCobaltblood) [reply code/modules/clothing/suits/hoodies.dm]: How can this be done alternatively? I just referenced how RemoveHood() works, but adjusted it for these icon states.

File: code/modules/clothing/suits/hoodies.dm (hunk 7 above)
Comment (VTCobaltblood) [reply code/modules/clothing/suits/hoodies.dm]: cuz i forgot to change it smh

File: code/modules/clothing/suits/hoodies.dm (hunk 8 above)
Comment (Karolis2011) [reply code/modules/clothing/suits/hoodies.dm]:
```DM
icon_open = initial(icon_open)
icon_closed = initial(icon_closed)```

Comment (VTCobaltblood) [reply code/modules/clothing/suits/hoodies.dm]: What's the difference?

**Pull request merged**